### PR TITLE
batter (upwardly)

### PR DESCRIPTION
**Pull request by horrible-little-slime**

### CODE REVIEW

File: src/extrovermectin.ts [hunk 1]
```diff
@@ -208,19 +208,33 @@ function initializeDireWarren(): void {
...
+    }).maximize();
+
+    do {
+      adventureMacro(
+        $location`The Dire Warren`,
+        Macro.if_($monster`fluffy bunny`, Macro.skill($skill`Batter Up!`)).step(embezzlerMacro())
+      );
+    } while ("fluffy bunny" !== get("lastEncounter"));
+  } else {
+    const banish = options.sort((a, b) => mallPrice(a) - mallPrice(b))[0];
```
[the remark this answers is not in the document]
Comment (horrible-little-slime) [reply src/extrovermectin.ts:229]: One thing to note is that those are only options for the orbless, who presumably aren't huffing as many extros. It's a legitimate concern regardless, but one that's out of scope of this PR.

**Comment by horrible-little-slime** [comment]
Right now this requires a mafia change to fix maximizer behavior.

File: src/extrovermectin.ts [hunk 2]
```diff
@@ -208,19 +220,46 @@ function initializeDireWarren(): void {
   if (options.some((option) => banishedMonsters.get(option) === $monster`fluffy bunny`)) {
     return;
   }
-  if (banishedMonsters.get($skill`Furious Wallop`) === $monster`fluffy bunny`) return;
+  if (banishedMonsters.get($skill`Batter Up!`) === $monster`fluffy bunny`) return;
```
Comment (docrostov) [inline src/extrovermectin.ts:223]: q: what did the line cook at waffle house say to his buddy when he finished mixing the waffle stuff

a: "batter up!"

i am personally illegal in at least 26 states

File: src/extrovermectin.ts [hunk 3]
```diff
@@ -208,19 +220,46 @@ function initializeDireWarren(): void {
...
-  acquire(1, banish, 50000, true);
-  do {
-    adventureMacro(
-      $location`The Dire Warren`,
-      Macro.if_($monster`fluffy bunny`, Macro.item(banish)).step(embezzlerMacro())
-    );
-  } while ("fluffy bunny" !== get("lastEncounter"));
+  const canBat = myFury() >= 5 && have($skill`Batter Up!`);
+  if (canBat) {
+    if (have($skill`Iron Palm Technique`) && !have($effect`Iron Palms`)) {
```
Comment (docrostov) [inline src/extrovermectin.ts:230]: this is inverted logic. iron palms screws up clubs, you explicitly do not want iron palms on if you want to cast batter up.

File: src/extrovermectin.ts [hunk 4]
```diff
@@ -208,19 +220,46 @@ function initializeDireWarren(): void {
...
+      useSkill($skill`Iron Palm Technique`);
+    }
+    const availableClub =
+      Item.all().find(
+        (i) =>
+          have(i) &&
+          canEquip(i) &&
+          weaponHands(i) === 2 &&
+          (itemType(i) === "club" || (have($effect`Iron Palms`) && itemType(i) === "sword"))
```
Comment (docrostov) [inline src/extrovermectin.ts:239]: please test this but the last time i tried to batter up with an iron palm'd sword it did not work. it is possible they changed this due to the bug reports tho?

**Review by docrostov — COMMENTED** [review]
i think this is pretty good but please test the palms interaction

File: src/extrovermectin.ts (hunk 4 above)
Comment (horrible-little-slime) [reply src/extrovermectin.ts:239]: https://cdn.discordapp.com/attachments/864070665615179786/1008239913025671199/unknown.png

**Pull request merged**